app/value_objects/conviction_type.rb

```diff
@@ -62,7 +62,7 @@ def initialize(raw_value, params = {})
     DISMISSAL                          = new(:dismissal,                        parent: MILITARY, skip_length: true, calculator_class: Calculators::AdditionCalculator::StartPlusSixMonths),
     OVERSEAS_COMMUNITY_ORDER           = new(:overseas_community_order,         parent: MILITARY, calculator_class: Calculators::AdditionCalculator::PlusSixMonths),
     SERVICE_COMMUNITY_ORDER            = new(:service_community_order,          parent: MILITARY, calculator_class: Calculators::AdditionCalculator::PlusSixMonths),
-    SERVICE_DETENTION                  = new(:service_detention,                parent: MILITARY, skip_length: true, calculator_class: Calculators::AdditionCalculator::StartPlusSixMonths),
+    SERVICE_DETENTION                  = new(:service_detention,                parent: MILITARY, skip_length: false, calculator_class: Calculators::AdditionCalculator::PlusSixMonths),
```

Comment:
You can remove the `skip_length` key because the default is false. Only the exceptions will set it to `true`.